### PR TITLE
Swap float should using Single-Precision Floating-Point instr 

### LINKED ISSUE #247: [HIFIVE]  cctest/test-code-generator/FuzzAssembleMoveAndSwap failed
**Describe the bug**

**To Reproduce**
1. Build with options `riscv64.native.release`
2. Run test/program `out/riscv64.native.release/cctest test-code-generator/FuzzAssembleMoveAndSwap --random-seed=-442682909 --nohard-abort --testing-d8-test-runner`
3. See error

**Expected behavior**

**Errors/Logs**
```
=== cctest/test-code-generator/FuzzAssembleMoveAndSwap ===
#
# Fatal error in , line 0
# Expected: '0.21766 (kRepFloat32 register)' but got 'nan (kRepFloat32 register)'
#
#
#
#FailureMessage Object: 0x3fffb9b5e8
==== C stack trace ===============================

    /home/luyahan/v8-riscv64/out/riscv64.native.release/cctest(+0x202cf68) [0x2ace30af68]
    /home/luyahan/v8-riscv64/out/riscv64.native.release/cctest(+0x1fa8394) [0x2ace286394]
    /home/luyahan/v8-riscv64/out/riscv64.native.release/cctest(+0x1e81d4a) [0x2ace15fd4a]
    /home/luyahan/v8-riscv64/out/riscv64.native.release/cctest(+0x8a6f7c) [0x2accb84f7c]
    /home/luyahan/v8-riscv64/out/riscv64.native.release/cctest(+0x888098) [0x2accb66098]
    /home/luyahan/v8-riscv64/out/riscv64.native.release/cctest(+0x80f666) [0x2accaed666]
    /lib64/lp64d/libc.so.6(__libc_start_main+0xa6) [0x3fc284e204]
    /home/luyahan/v8-riscv64/out/riscv64.native.release/cctest(_start+0x2c) [0x2accb64ff8]
Command: out/riscv64.native.release/cctest test-code-generator/FuzzAssembleMoveAndSwap --random-seed=-442682909 --nohard-abort --testing-d8-test-runner
--- FAILED ---
```

**Additional context**

### PR DESCRIPTION
**Pull request by luyahan**
Fixes #247

### CODE REVIEW

File: src/compiler/backend/riscv64/code-generator-riscv64.cc [hunk 1]
```diff
@@ -2691,9 +2691,16 @@ void CodeGenerator::AssembleSwap(InstructionOperand* source,
       } else {
         DCHECK(destination->IsFPStackSlot());
         MemOperand dst = g.ToMemOperand(destination);
-        __ Move(temp, src);
-        __ LoadDouble(src, dst);
-        __ StoreDouble(temp, dst);
+        if (rep == MachineRepresentation::kFloat32) {
+          __ fmv_s(temp, src);
```
Comment (fw-brice) [inline src/compiler/backend/riscv64/code-generator-riscv64.cc:2695]: I'd prefer to use a higher level interface here and try to only call `Assembler` APIs from the `TurboAssembler`. I'd suggest either adding a third parameter to the floating point `Move` or creating a `MoveF` and a `MoveD`. 

@penguinwu, I think you did some refactoring of these moves before. Your thoughts?

**Review by fw-brice — CHANGES_REQUESTED** [review]
Please update your commit message formatting. Keep the title below 50 characters and use the proper formatting to indicate a fixed issue.

```
Use single precision instr to swap floats

Fixes #247 
```

File: src/compiler/backend/riscv64/code-generator-riscv64.cc (hunk 1 above)
Comment (penguinwu) [reply src/compiler/backend/riscv64/code-generator-riscv64.cc:2695]: Similar to `LoadDouble/StoreDouble`, we can add `MoveDouble` and `MoveFloat`.

**Review by fw-brice — APPROVED** [review]
LGTM

**Comment by fw-brice** [comment]
There are some failures in the CI job here that were not properly reported (I opened #256 for this), but they are not related to these changes, so I will merge it.

**Pull request merged**